mlflow/server/handlers.py

```diff
@@ -1103,6 +1104,117 @@ def _default_history_bulk_impl():
     }
 
 
+def _get_sampled_steps_from_steps(start_step, end_step, max_results):
+    num_steps = end_step - start_step + 1
+    interval = num_steps / max_results
+    grouped_steps = {}
+    # include both start_step & end_step
```

Comment:
Does this mean the result includes both `start_step & end_step`?

Reply:
Nope, this means we need to check both start_step & end_step. But in reality, both start_step and end_step will be included in the final steps used.